apps/css-workshop/src/components/Tag.astro

```diff
@@ -0,0 +1,41 @@
+---
+const { title, className, id, style, button = true, variant, href, ...props } = Astro.props;
+---
+
+<>
+  {
+    variant === 'basic' && !href && (
+      <span class='iui-tag-basic' id={id}>
+        {title}
+      </span>
+    )
+  }
+  {
+    href && (
+      <a href={href} class='iui-anchor iui-tag-basic' id={id}>
```

Comment:
Instead of inlining the `iui-anchor` class here, it would be useful to create an `Anchor.astro` component (in a separate PR), since the `iui-anchor` class belongs to that component.

Until then, I think `iui-anchor` should be passed externally in the place where it used. For example:

```astro
<Tag_ as="a" class="iui-anchor" href="#the-url">Click me</Tag_>
```

Reply:
So i've been having trouble with this. I've tried implementing the solution you suggested but in order to render the a tag properly I've ended up using almost identical logic to the original i.e. conditional branching based off props. When I don't the tag renders as a span with the `as='a'` attribute. Would you be able to clarify how to avoid the conditional branching? or was your comment soley related to the class and not to the branching?